xla/service/cpu/onednn_ops_rewriter.cc

```diff
@@ -125,29 +152,28 @@ std::optional<HloInstruction*> MatchSoftmax(HloInstruction* instr) {
 
   // Upper diamond
   if (!Match(left_exponential->mutable_operand(0),
-             m::Subtract(
-                 m::Op(&left_producer),
-                 m::Broadcast(
-                     m::Reshape(m::Broadcast(m::Reshape(
-                         m::Reduce(m::Op(&right_producer), m::Op())
-                             .WithPredicate([](const HloInstruction* reduce) {
-                               HloComputation* reducer = reduce->to_apply();
-                               return (reducer->root_instruction()->opcode() ==
-                                           HloOpcode::kMaximum &&
-                                       reduce->dimensions().size() == 1 &&
-                                       reduce->dimensions()[0] !=
-                                           reduce->shape().rank() - 1);
-                             })
-                             .WithOneUse()))))
-                     .WithOneUse())
+             m::Subtract(m::Op(&left_producer),
+                         m::Broadcast(m::Reshape(m::Broadcast(
+                                          m::Reshape(m::Op(&reduce_instr)))))
+                             .WithOneUse())
                  .WithOneUse())) {
     return std::nullopt;
   }
 
-  if (left_producer != right_producer || left_producer->user_count() != 2) {
+  // Match the reduce max
+  if (!Match(reduce_instr, MaxReduce(&reduce_max))) return std::nullopt;
+
+  if (left_producer != reduce_max->operand(0) ||
+      left_producer->user_count() != 2) {
     return std::nullopt;
   }
 
+  if ((reduce_sum->dimensions()[0]) != (reduce_max->dimensions()[0])) {
```

Comment:
nit: please remove the unnecessary parentheses

Reply:
Done